codegen/src/main/scala/bsp/codegen/IR.scala

```diff
@@ -0,0 +1,61 @@
+package bsp.codegen
+
+import software.amazon.smithy.model.shapes.ShapeId
```

Comment:
This file represents the "intermediate representation" that acts as a middleman between the smithy-model and the renderer. It contains the cherry-picked, sufficient information for re-creating the bsp4j code.